modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/NotificationPermissionHelper.kt

```diff
@@ -0,0 +1,39 @@
+package au.com.shiftyjelly.pocketcasts.utils
+
+import android.Manifest
+import android.app.Activity
+import android.content.pm.PackageManager
+import android.os.Build
+import androidx.activity.result.ActivityResultLauncher
+import androidx.core.content.ContextCompat
+
+object NotificationPermissionHelper {
+
+    fun checkForNotificationPermission(
+        activity: Activity,
+        launcher: ActivityResultLauncher<String>,
+        onShowRequestPermissionRationale: () -> Unit = {},
+        onPermissionGranted: () -> Unit = {},
+        onPermissionHandlingNotRequired: () -> Unit = {},
```

Comment:
Originally I put just one callback, the onPermissionGranted one. But I noticed that in MainActivity we were not doing anything for the case we don't need to require permission when call `checkForNotificationPermission`, so I decided to do this to avoid any potential issue